### ISSUE TITLE
Proposed solution to: access to dashboard from public (i.e. no user) gives wrong data representation respect to user access

### ISSUE THREAD

**Issue by And2IT** [emoncms/emoncms#1823]
Preamble: Emoncms installed on Raspberry, used in a timezone different from UTC. Only one user created in Emoncms, with correct timezone setting (for example: Europe/Rome).
What happens: when viewing a dashboard from logged user, date is correctly represented. When viewing the same dashboard from a public access (i.e. without logged user), date is wrong and also data are not correctly represented.

![immagine](https://user-images.githubusercontent.com/116438365/200812401-d91baade-8eec-41b0-9669-a10cfe1384be.png)

On the left, dashboard viewed after having logged in; on the right, the same dashboard viewed without having logged in.

I have resolved -temporarily- editing the file feed_controller.php, row 31:

  `      $user_timezone = 'Europe/Rome';`

instead of the original:

`        $user_timezone = 'UTC';`

I kindly ask to modify the master branch to apply the timezone of the emoncms user, also in case of public access to dashboards.

Thank you.

**Comment by Jaddache** [comment]
> I am experiencing the same issue and the proposed change above has the graph displaying correctly now - although I now have a local change to the repository....
I also request the master branch to be modified as per @And2IT request above.

**Comment by TrystanLea** [comment]
yes agreed a long standing issue, see: https://github.com/emoncms/emoncms/issues/1060 no immediate plan to work on this yet but it has been noted.

**Comment by adelriop** [comment]
I also have this issue and don't know how to solve it. If I connect from my phone in my timezone to a public dashboard on emoncms.org it shows the right feed time stamp. However, I have an old kindle showing a screenshot of that dashboard. The screenshot is taken in a heroku app and it shows one hour less. I hope it get fixed someday.

**Comment by And2IT** [comment]
> I also have this issue and don't know how to solve it. If I connect from my phone in my timezone to a public dashboard on emoncms.org it shows the right feed time stamp. However, I have an old kindle showing a screenshot of that dashboard. The screenshot is taken in a heroku app and it shows one hour less. I hope it get fixed someday.

You can temporary solve it with the proposed code modification as in my first post here.

**Comment by And2IT** [comment]
@borpin 
Can you help me/us in pushing more effort to resolve the issue ?
Thanks !
Andrea

**Comment by adelriop** [comment]
> You can temporary solve it with the proposed code modification as in my first post here.

I have my dashboard hosted by https://emoncms.org/ so I guess that change on the code is something that I cannot do and it would only be valid if I hosted my own emoncms, right? Thanks!

**Comment by chaveiro** [comment]
Public dashboards use the browser location timezone. Of that is the case, no bug exist.

**Comment by adelriop** [comment]
> Public dashboards use the browser location timezone. Of that is the case, no bug exist.
> Ensure device has the right region defined and use a supported browser.

As I mentioned on my post, I have a heroku app that takes the screenshot of the dashboard so show on a kindle device. I was not able to change the timezone of the heroku instance. The timezone of the heroku server that takes the screenshot is on a different timezone. I would like my dashboard to show the same time always -> https://emoncms.org/dashboard/view?id=62642

If you have any workaround, please let me know! Thanks.

**Comment by chaveiro** [comment]
Pass the readkey of the user on the url so it gets logged in.

**Comment by And2IT** [comment]
> Public dashboards use the browser location timezone. If that is the case, no bug exist.
> 
> Ensure device has the right region defined and use a supported browser.

As of my knowledge, having tested the locally installed emoncms on a Raspberry, the bug exists. In my install, the Raspberry (where Emoncms is installed) and the web browser are both in the same timezone (Europe/Rome) and it doen't work correctly when access is performed with no account. The only way to solve is hacking the code as told in first post.
@TrystanLea confirms the bug exists.

**Comment by adelriop** [comment]
> Pass the readkey of the user on the url so it gets logged in.

I could not make it work. But I found a way to change the timezone of my heroku app and now it is working! Thanks!
